edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Core.Test/metrics/AvaliabilityMetricsTest.cs

```diff
@@ -0,0 +1,241 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.Core.Test.Metrics
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Akka.Event;
+    using App.Metrics;
+    using Microsoft.Azure.Devices.Edge.Agent.Core.Metrics;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Azure.Devices.Edge.Util.Metrics;
+    using Microsoft.Azure.Devices.Edge.Util.Metrics.NullMetrics;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Microsoft.Extensions.Logging.Abstractions;
+    using Moq;
+    using Xunit;
+
+    [Unit]
+    public class AvailabilityMetricsTest : TempDirectory
+    {
+        static readonly TestConfig Config = new TestConfig("image1");
+        static readonly ConfigurationInfo DefaultConfigurationInfo = new ConfigurationInfo("1");
+        static readonly IDictionary<string, EnvVal> EnvVars = new Dictionary<string, EnvVal>();
+
+        [Fact]
+        public void ReportsAccurateAvailability()
```

Comment:
Can you add a brief description of what the test is trying to test and what's the expected result for each of the test please?